knowledge/guidelines.rst

```diff
@@ -0,0 +1,50 @@
+Core guidelines
+===============
+
+
+Good practices
+--------------
+
+- The recipes' ``generate()`` method purpose is to prepare the build as much as possible.
```

Comment:
Maybe we can reference the [tutorial part for the generate](https://docs.conan.io/en/2.0/tutorial/creating_packages/preparing_the_build.html) here? Also, I think we can add some notes in each section, like a note of good practices for each part of the documentation. But we can do that in the future starting from this guide

Reply:
I don't want to distract too much from this guidelines, driving users to different places in the docs, but this is like a "checklist" to go through, making sure you are following them. If you need to understand what the ``generate()`` method is, maybe it is too early to understand the "good practices"?
(This is a bit extreme opinion, I am fine with having some links, as long as it is not cluttered with many of them, that it appears more like an index to continue navigation than a checklist)

I have already started to add them in different places in the docs, like https://docs.conan.io/en/2.0/reference/commands/cache.html (scroll to the bottom)

Reply:
> but this is like a "checklist" to go through, making sure you are following them. If you need to understand what the generate() method is, maybe it is too early to understand the "good practices"?

This idea and belief should be the first sentence on the page so people know what to expect and set them up for success 🙏